nautobot/utilities/tables.py

```diff
@@ -88,26 +89,31 @@ def __init__(self, *args, user=None, **kwargs):
                     self.columns.hide(column.name)
 
         # Apply custom column ordering for user
+        columns = []
+        pk = self.base_columns.pop("pk", None)
+        actions = self.base_columns.pop("actions", None)
+
         if user is not None and not isinstance(user, AnonymousUser):
             columns = user.get_config(f"tables.{self.__class__.__name__}.columns")
-            if columns:
-                pk = self.base_columns.pop("pk", None)
-                actions = self.base_columns.pop("actions", None)
-
-                for name, column in self.base_columns.items():
-                    if name in columns:
-                        self.columns.show(name)
-                    else:
-                        self.columns.hide(name)
-                self.sequence = [c for c in columns if c in self.base_columns]
-
-                # Always include PK and actions column, if defined on the table
-                if pk:
-                    self.base_columns["pk"] = pk
-                    self.sequence.insert(0, "pk")
-                if actions:
-                    self.base_columns["actions"] = actions
-                    self.sequence.append("actions")
+        if columns:
+            for name, column in self.base_columns.items():
+                if name in columns:
+                    self.columns.show(name)
+                else:
+                    self.columns.hide(name)
+            self.sequence = [c for c in columns if c in self.base_columns]
+
+        # Always include PK and actions column, if defined on the table
+        if pk:
+            with contextlib.suppress(ValueError):
```

Comment:
Why not `if "pk" in self.sequence`, same with below?

Reply:
That's the pattern in `develop` and `next`, I agree tho it might be over-engineering but it is working for now.